code/go/0chain.net/sharder/blockstore/db.go

```
@@ -35,128 +30,63 @@ const (
-)
-
-//Create db file and create buckets
-func InitMetaRecordDB(bmrDB, qmrDB string, deleteExistingDB bool) {
-	wg := sync.WaitGroup{}
-	wg.Add(2)
-
-	if deleteExistingDB {
-		os.Remove(bmrDB)
-		os.Remove(qmrDB)
-	}
-	//Open db for storing whereabout of blocks
-	go func() {
-		defer wg.Done()
-		var err error
-		parentDir, _ := filepath.Split(bmrDB)
-		if err := os.MkdirAll(parentDir, 0644); err != nil {
-			panic(err)
-		}
+	DefaultHostRedisDB     = "redis"
+	DefaultPortRedisDB     = "6379"
+	DefaultPasswordRedisDB = ""
+	DefaultNumberRedisDB   = 0
 
-		bwrDB, err = bbolt.Open(bmrDB, 0644, bbolt.DefaultOptions) // fix me
-		if err != nil {
-			panic(err)
-		}
+	redisHashCacheHashAccessTime      = "redisHashCacheHashAccessTime"
+	redisSortedSetCacheAccessTimeHash = "redisSortedSetCacheAccessTimeHash"
+	redisSortedSetUnmovedBlock        = "redisSortedSetUnmovedBlock"
 
-		err = bwrDB.Update(func(t *bbolt.Tx) error {
-			_, err := t.CreateBucketIfNotExists([]byte(BlockWhereBucket))
-			if err != nil {
-				return err
-			}
-			_, err = t.CreateBucketIfNotExists([]byte(UnmovedBlockBucket))
-			return err
-		})
-
-		if err != nil {
-			panic(err)
-		}
-
-	}()
-
-	go func() {
-		defer wg.Done()
-		var err error
-		parentDir, _ := filepath.Split(qmrDB)
-		if err := os.MkdirAll(parentDir, 0644); err != nil {
-			panic(err)
-		}
+	CacheAccessTimeSeparator = ":"
+)
 
-		qDB, err = bbolt.Open(qmrDB, 0644, bbolt.DefaultOptions) // fix me
-		if err != nil {
-			panic(err)
-		}
+// InitMetaRecordDB Create db file and create buckets.
+func InitMetaRecordDB(host, port, password string, numDB int, deleteExistingDB bool) {
 
-		err = qDB.Update(func(t *bbolt.Tx) error {
-			_, err = t.CreateBucketIfNotExists([]byte(BlockUsageBucket))
-			return err
-		})
-	}()
+	redisClient = redis.NewClient(&redis.Options{
+		Addr:     host + ":" + port,
+		Password: password,
+		DB:       numDB,
+	})
 
-	wg.Wait()
+	if deleteExistingDB {
+		_, _ = redisClient.FlushDB().Result()
```

If any error is there we can simply not start sharder. Let's not ignore error here.